### ISSUE TITLE
Unneeded parentheses around build numbers since 4.0.3

### ISSUE THREAD

**Issue by yan12125** [buildbot/buildbot#8046]
For example, on https://buildbot.buildbot.net/#/console, each build link has texts with parentheses:
![圖片](https://github.com/user-attachments/assets/21f62f88-5d24-4797-bffb-1d090934423b)

On 4.0.2, there are no such parentheses, as shown in https://buildbot.python.org/#/console:
![圖片](https://github.com/user-attachments/assets/d19ee73e-20d8-459f-8573-dc6bb58e151b)

This is probably a regression since https://github.com/buildbot/buildbot/pull/7613, where parentheses are used unconditionally.

**Comment by classabbyamp** [comment]
i just added this to my config: `c['www']['ui_default_config'] = { 'Links.build_link_template': "%(build_number)", }`

**Comment by yan12125** [comment]
Thanks, but the branch name is useful in some places. For example, https://buildbot.buildbot.net/#/builders shows what branches are run for each builder, and the information is gone with `%(build_number)`.

**Comment by p12tic** [comment]
Yeah, turns out build link template needs to support proper logic.
